pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -425,12 +425,19 @@ def _map_tool_definition(f: ToolDefinition) -> BetaToolParam:
 
 
 def _map_usage(message: BetaMessage | BetaRawMessageStreamEvent) -> usage.Usage:
-    if isinstance(message, BetaMessage):
-        response_usage = message.usage
-    elif isinstance(message, BetaRawMessageStartEvent):
-        response_usage = message.message.usage
-    elif isinstance(message, BetaRawMessageDeltaEvent):
-        response_usage = message.usage
+    """Maps Anthropic API message object to pydantic-ai Usage object, extracting integer-type usage statistics.
+
+    Handles BetaMessage, BetaRawMessageStartEvent, BetaRawMessageDeltaEvent.
+    Returns empty Usage if type doesn't contain usage info.
+    """
+    msg_type = type(message)
+
+    if msg_type is BetaMessage:
+        response_usage = cast(BetaMessage, message).usage
+    elif msg_type is BetaRawMessageStartEvent:
+        response_usage = cast(BetaRawMessageStartEvent, message).message.usage
+    elif msg_type is BetaRawMessageDeltaEvent:
+        response_usage = cast(BetaRawMessageDeltaEvent, message).usage
```

Comment:
Are the performance improvement metrics in the description still accurate after the latest changes to this PR?

I think this change reduces readability (and consistency with other code) significantly so I'd only want to do it if the performance impact (on real examples, not just the `test_usage` test) are significant. And if it is significant, should we do this everywhere we currently use `isinstance` (and don't care about subclasses)? Otherwise it seems odd to do it only in this specific place.